physics/module_sf_mynn.F90

```diff
@@ -344,7 +345,12 @@ SUBROUTINE SFCLAY_mynn(                           &
      &                     qsfc_wat,  qsfc_lnd,  qsfc_ice
 
 ! CCPP error handling
+#ifndef _OPENACC
       character(len=*), intent(inout) :: errmsg
+#else
```

Comment:
This looks fine for non-GPU applications, but I doubt that this works when running the physics within a host w/ GPU support enabled? 
errmsg is defined by the host, and used throughout the physics, so changing it in this one place is not sufficient, you would need to conditionally change it everywhere. At least for this to work inline.

Reply:
Hi Dustin. I'm pushing a new change up to this branch which removes these preprocessor substitutions in favor of a local variable approach with explicit extent as you previously suggested. However, this does unfortunately add new complexity due to the use of return statements mid loop that have to be handled specially, so 4 new local variables were necessary to handle this. Please see the commit message for full details explaining this.

Also, note that some preprocessor directives are still unavoidable using __OPENACC despite this change. However, __OPENACC is set by the compiler automatically any time OpenACC is enabled at compile time, so CCPP need not track this.